pages/reset-password/index.tsx

```diff
@@ -1,22 +1,153 @@
-import Input from "@/components/primitives/Form/Input";
 import Image from "next/image";
-import React, { useState } from "react";
-import { useRouter } from "next/navigation";
+import React, { useEffect, useState } from "react";
+import { useRouter, NextRouter } from "next/router";
 import Link from "next/link";
+import Input from "@/components/primitives/Form/Input";
+import Button from "@/components/primitives/Button/button";
+import { useForm, SubmitHandler } from "react-hook-form";
+import { z, ZodError, ZodType } from "zod";
+
+// Define custom API error type
+type ApiErrorResponse = {
+  status: "Error";
+  message: string;
+};
+
+interface FormValues {
+  password: string;
+  confirmPassword: string;
+}
+
+type formData = {
+  password: string;
+  confirmPassword: string;
+};
 
-export default function ResetPassword() {
-  const router = useRouter();
-  const [values, setValues] = useState({
-    password: "",
-    confirmPassword: "",
+interface QueryParams {
+  token?: string;
+}
+
+// Define Zod schema for password and confirmPassword validation
+const passwordSchema: ZodType<formData> = z
+  .object({
+    password: z
+      .string()
+      .min(8, {
+        message: "Password must must not be less than 8 characters",
+      })
+      .refine((value) => /[!@#$%^&*(),.?":{}|<>]/.test(value), {
+        message: "Password must contain at least one special character",
+      })
+      .refine((value) => /\d/.test(value), {
+        message: "Password must contain at least one digit",
+      })
+      .refine((value) => /[a-z]/.test(value), {
+        message: "Password must contain at least one lowercase letter",
+      })
+      .refine((value) => /[A-Z]/.test(value), {
+        message: "Password must contain at least one uppercase letter",
+      }),
+    confirmPassword: z.string(),
+  })
+  .refine((data) => data.password === data.confirmPassword, {
+    message: "Passwords do not match",
+    path: ["confirmPassword"],
   });
 
-  const submitForm = (e: React.FormEvent<HTMLFormElement>) => {
-    e.preventDefault();
-    router;
-    values;
+const ResetPassword: React.FC = () => {
+  const router: NextRouter = useRouter();
+  const [serverError, setServerError] = useState("");
+  const { token }: QueryParams = router.query as QueryParams;
+
+  const {
+    handleSubmit,
+    register,
+    formState: { errors: formErrors },
+    setError,
+    clearErrors,
+  } = useForm<FormValues>({});
```

Comment:
I don't see you using the zod resolver for react-hook-form. Can you please explain how you're zod to validate the form input?